sphinx/addnodes.py

```diff
@@ -342,6 +342,51 @@ class pending_xref(nodes.Inline, nodes.Element):
     """
 
 
+class pending_xref_condition(nodes.Inline, nodes.TextElement):
```

Comment:
@shimizukawa @jakobandersen What do you think about this new node?

Reply:
@jakobandersen Please let me know your opinion on this new node. I don't know this is worthy for C/C++ domains. But I  believe this will not conflict with them.